### PR TITLE
Created pilatus detector base classes

### PR DESCRIPTION
**Pull request by mskinner5278**

### CODE REVIEW

**Comment by JunAishima** [comment]
please see setup.py in mxtools for registering the new handler - NSLS-II-AMX/mxtools@5658441c7192748ca6be6f5a1fa794fe0faecbd2

File: nyxtools/handlers.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class PilatusHandlerMX(HandlerBase):
+    spec = "AD_PILATUS_MX"
+
+    def __init__(self, fpath, seq_id):
+        self._seq_id = seq_id
+        self._fpath = pathlib.Path(f"{fpath}_{seq_id}.cbf").absolute()
+        if not self._fpath.is_file():
+            raise RuntimeError(f"File {self._fpath} does not exist")
+
+    def __call__(self, data_key="data", **kwargs):
```
Comment (tacaswell) [inline nyxtools/handlers.py:19]: Why are you taking `**kwargs` if they are not used or passed on?

File: nyxtools/pilatus.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class PilatusSimulatedFilePlugin(Device, FileStoreBase):
+    file_path = ADComponent(EpicsPathSignal, "FilePath", string=True, path_semantics="posix")
+    external_name = Cpt(Signal, value="")
+
+    def __init__(self, *args, **kwargs):
+        self.filestore_spec = "AD_PILATUS_MX"
+        super().__init__(*args, **kwargs)
+
+    def stage(self):
+        write_path = datetime.datetime.now().strftime(self.write_path_template)
```
Comment (tacaswell) [inline nyxtools/pilatus.py:17]: This is a local variable that is dropped when we lease the frame, did you mean to stash on the object and have it be used later?

File: nyxtools/pilatus.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return super().generate_datum(key, timestamp, datum_kwargs)
+
+class PilatusBase(PilatusDetector):
+    file = Cpt(PilatusSimulatedFilePlugin,
+               suffix="cam1:",
+               write_path_template="",
+               root="")
+    image = Cpt(ImagePlugin, "image1:")
+
+    def stage(self, *args, **kwargs):
```
Comment (tacaswell) [inline nyxtools/pilatus.py:32]: If you are only calling `super()` there is no need for these methods.

File: nyxtools/pilatus.py (hunk 2 above)
Comment (mskinner5278) [reply nyxtools/pilatus.py:17]: Yes, I am still trying to understand how the staging worked for the Eiger plugin so I can adapt it for this one. There are still pieces missing that I am filling in.

File: nyxtools/handlers.py (hunk 1 above)
Comment (mskinner5278) [reply nyxtools/handlers.py:19]: I am addressing these issues now, thank you

File: nyxtools/pilatus.py (hunk 3 above)
Comment (mskinner5278) [reply nyxtools/pilatus.py:35]: I have left two empty methods in for now while we develop them. But they are just reminders until I add something to them.

**Pull request merged**